### ISSUE TITLE
[Bug]not working properly in Canvas mode, duplicate last letter

### ISSUE THREAD

**Issue by NTUYi** [artisticat1/obsidian-latex-suite#123]
This plugin does not work properly in canvas mode, such as when typing `mk`, it becomes `$k$` instead of `$$`. I have tested on both Obsidian 1.1.14 and 1.1.9 with Latex Suite 1.6.12

**Comment by VoidR** [comment]
me too. with Obsidian 1.1.16 and Latex Suite 1.6.12.

**Comment by VoidR** [comment]
> me too. with Obsidian 1.1.16 and Latex Suite 1.6.12.

I reduced the Latex Suite version to 1.6.10 to solve this problem

**Comment by VoidR** [comment]
> I can't reproduce this. Does this happen all the time? Does it happen in the sandbox vault / with other plugins disabled? Could you paste your debug info here (Command palette -> Show debug info)?

SYSTEM INFO:
	Obsidian version: v1.1.16
	Installer version: v0.15.9
	Operating system: Darwin Kernel Version 22.3.0: Mon Jan 30 20:39:35 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T8103 22.3.0
	Login status: logged in
	Catalyst license: none
	Insider build toggle: off
	Live preview: on
	Legacy editor: off
	Base theme: light
	Community theme: none
	Snippets enabled: 0
	Restricted mode: off
	Plugins installed: 26
	Plugins enabled: 19
		1: Citations v0.4.5
		2: Dataview v0.5.55
		3: Calendar v1.5.10
		4: Advanced Tables v0.18.1
		5: Outliner v4.2.1
		6: QuickAdd v0.11.6
		7: Hider v1.2.4
		8: 盘古 PanGu v1.6.1
		9: Admonition v9.2.1
		10: Sortable v0.3.1
		11: Day Planner v0.5.8
		12: Templater v1.16.0
		13: Zotero Integration v2.3.6
		14: Style Settings v1.0.3
		15: Hover Editor v0.11.9
		16: Contextual Typography v2.2.4
		17: Minimal Theme Settings v6.3.1
		18: Homepage v2.8
		19: Latex Suite v1.6.12

RECOMMENDATIONS:
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.

**Comment by artisticat1** [comment]
OK, thanks!

- Does the issue still occur when using Latex Suite 1.6.11?
- Does this happen all the time, or only after expanding certain snippets?
- Does this happen in the sandbox vault with all other plugins disabled?

**Comment by VoidR** [comment]
> OK, thanks!
> 
> * Does the issue still occur when using Latex Suite 1.6.11?
> * Does this happen all the time, or only after expanding certain snippets?
> * Does this happen in the sandbox vault with all other plugins disabled?

* yes, the issue still occur when using Latex Suite 1.6.11.
* maybe, it happened with using canvas.
* no, I did't try sandbox vault.

**Comment by VoidR** [comment]
> OK, thanks!
> 
> * Does the issue still occur when using Latex Suite 1.6.11?
> * Does this happen all the time, or only after expanding certain snippets?
> * Does this happen in the sandbox vault with all other plugins disabled?

When this issue disappears, another issue will appear, when using the keyboard to control the cursor movement, it will move two characters at a time. These two issues conflict with each other.

**Comment by artisticat1** [comment]
> yes, the issue still occur when using Latex Suite 1.6.11.

OK, thank you!

> no, I did't try sandbox vault.

Could you try it and let me know whether the issue still occurs?

> maybe, it happened with using canvas.

Does the issue only occur inside canvases, or also in markdown notes? Do you see the same error in the console as @nsmoore57?

**Comment by artisticat1** [comment]
Should be fixed in the latest version.
